src/providers/MeetingProvider/MeetingManager.ts

```diff
@@ -601,59 +615,37 @@ export class MeetingManager implements AudioVideoObserver {
         innerDevice = device;
       }
 
-      if (innerDevice === null) {
-        this.selectedAudioInputDevice = null;
-      } else if (typeof innerDevice === 'string') {
-        this.selectedAudioInputDevice = innerDevice;
-      } else if (innerDevice instanceof MediaStream) {
-        this.selectedAudioInputDevice = innerDevice.id;
-      } else {
-        const deviceId = this.getDeviceID(innerDevice);
+      const deviceId =
+        DefaultDeviceController.getIntrinsicDeviceId(innerDevice);
+      if (typeof deviceId === 'string') {
         this.selectedAudioInputDevice = deviceId;
+      } else if (Array.isArray(deviceId) && deviceId[0]) {
+        this.selectedAudioInputDevice = deviceId[0];
       }
-
       this.selectedAudioInputTransformDevice = device;
     }
     this.publishSelectedAudioInputDevice();
     this.publishSelectAudioInputDeviceError();
     this.publishSelectedAudioInputTransformDevice();
   };
 
-  private getDeviceID = (
-    innerDevice: MediaTrackConstraints | MediaTrackConstraintSet
-  ): string => {
-    if (Array.isArray(innerDevice)) {
-      innerDevice = innerDevice[0];
-    }
-
-    const deviceId = innerDevice.deviceId;
-    if (typeof deviceId === 'string') {
-      return deviceId;
-    }
-    if (Array.isArray(deviceId)) {
-      return deviceId[0];
-    }
-    if (Array.isArray(deviceId?.exact) && deviceId?.exact) {
-      return deviceId.exact[0];
-    }
-    if (typeof deviceId?.exact === 'string') {
-      return deviceId.exact;
-    }
-    return '';
-  };
-
   selectAudioOutputDevice = async (deviceId: string): Promise<void> => {
     try {
       await this.audioVideo?.chooseAudioOutputDevice(deviceId);
       this.selectedAudioOutputDevice = deviceId;
       this.publishSelectedAudioOutputDevice();
     } catch (error) {
-      console.error(`Error setting audio output - ${error}`);
+      console.error(`Error setting audio output`, error);
     }
   };
 
-  selectVideoInputDevice = async (deviceId: string): Promise<void> => {
-    const receivedDevice = videoInputSelectionToDevice(deviceId);
+  selectVideoInputDevice = async (
```

Comment:
We also have to update the `useSelectVideoInput` hook to receive these parameters I think as that hook is a wrapper around what this method is, but that can be done as a follow up I guess.
https://aws.github.io/amazon-chime-sdk-component-library-react/?path=/story/sdk-hooks-useselectvideoinputdevice--page

Reply:
Yeah, I guess that hook will only be used for non-bg blur video devices for now.